fdb-relational-jdbc/src/test/java/com/apple/foundationdb/relational/jdbc/JDBCAutoCommitTest.java

```diff
@@ -80,6 +80,65 @@ void autoCommitOn() throws SQLException, IOException {
         }
     }
 
+    @Test
+    void autoCommitWithNoTransactionInBetween() throws SQLException {
+        try (RelationalConnection connection = DriverManager.getConnection(getTestDbUri()).unwrap(RelationalConnection.class)) {
+            connection.setAutoCommit(false);
+            connection.setAutoCommit(true);
+        }
+    }
+
+    @Test
+    void autoCommitStayOn() throws SQLException {
+        try (RelationalConnection connection = DriverManager.getConnection(getTestDbUri()).unwrap(RelationalConnection.class)) {
+            connection.setAutoCommit(true);
+        }
+    }
+
+    @Test
+    void autoCommitStayOff() throws SQLException {
+        try (RelationalConnection connection = DriverManager.getConnection(getTestDbUri()).unwrap(RelationalConnection.class)) {
+            connection.setAutoCommit(false);
+            connection.setAutoCommit(false);
+        }
+    }
+
+    @Test
+    void commitEnableAutoCommit() throws SQLException {
+        try (RelationalConnection connection = DriverManager.getConnection(getTestDbUri()).unwrap(RelationalConnection.class)) {
+            connection.setAutoCommit(false);
+
+            try (RelationalStatement statement = connection.createStatement()) {
+                insertOneRow(statement);
+            }
+            connection.commit();
+            connection.setAutoCommit(true);
+            try (RelationalStatement statement = connection.createStatement()) {
+                RelationalResultSet resultSet = statement.executeQuery("SELECT * FROM test_table");
```

Comment:
Maybe this would be more exhaustive with another insert after the setAutoCommit(true)?